app/src/main/kotlin/com/looker/droidify/ui/appList/AppListFragment.kt

```diff
@@ -82,9 +91,65 @@ class AppListFragment() : Fragment(), CursorOwner.Callback {
             isMotionEventSplittingEnabled = false
             setHasFixedSize(true)
             recycledViewPool.setMaxRecycledViews(AppListAdapter.ViewType.PRODUCT.ordinal, 30)
-            recyclerViewAdapter = AppListAdapter(source) {
-                screenActivity.navigateProduct(it.packageName)
+
+            var favouriteApps: Set<String> = emptySet()
+
+            lifecycleScope.launch {
+                favouriteApps = viewModel.settingsRepository.getInitial().favouriteApps
             }
+
+            recyclerViewAdapter = AppListAdapter(
+                source,
+                favouriteApps,
+                { screenActivity.navigateProduct(it.packageName) },
+                fun(productItem, menuItem): Boolean {
+
+                    when (menuItem.title) {
+
+                        getString(com.looker.core.common.R.string.install) -> lifecycleScope.launch {
```

Comment:
@Iamlooker I am unable to get the install to work, and I am pretty sure there's a better way around. can you point me in the right direction?

Reply:
You will need to create a connection with downloadService like [this](https://github.com/Droid-ify/client/blob/1687650a4b09febd368ff7737d0df701aa0e2e5c/app/src/main/kotlin/com/looker/droidify/ui/appDetail/AppDetailFragment.kt#L102) and then call the download function to start download service like [this](https://github.com/Droid-ify/client/blob/main/app/src/main/kotlin/com/looker/droidify/ui/appDetail/AppDetailFragment.kt#L345C16-L345C34)

I know the arch. is not the best(its genuinely bad) but I am working on it

Reply:
Sorry for late reply